### ISSUE TITLE
ServletApiContext.getResourcePaths() doesn't respect the spec

### ISSUE THREAD

**Issue by ppodgorsek** [eclipse/jetty.project#10084]
**Jetty version(s)**

12.0.0.beta3

**Java version/vendor** `(use: java -version)`

openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment (Red_Hat-17.0.7.0.7-4.fc38) (build 17.0.7+7)
OpenJDK 64-Bit Server VM (Red_Hat-17.0.7.0.7-4.fc38) (build 17.0.7+7, mixed mode, sharing)

**OS type/version**

Fedora 38

**Description**

I am trying to list files and directories within a `WEB-INF/` folder which has a structure similar to the one below:

```
/WEB-INF/
  lib/
  custom-taglibs/
    my-tablib.tld
  web.xml
```

According to the spec, a call to `servletApiContext.getResourcePaths(/WEB-INF/)` should return:

* /WEB-INF/lib/
* /WEB-INF/custom-taglibs/
* /WEB-INF/web.xml

Instead, the method returns directory names without a trailing `/`:

* /WEB-INF/lib
* /WEB-INF/custom-taglibs
* /WEB-INF/web.xml

This makes it much harder to determine which entries are subdirectories and which ones aren't.

**How to reproduce?**

Create subdirectories in an application's `/WEB-INF/` folder and call `ServletApiContext.getResourcePaths("/WEB-INF/")`.

**Comment by joakime** [comment]
Interesting...

The Servlet TCK doesn't actually test the Strings in the Set returned from `getResourcePaths(String)`

https://github.com/jakartaee/platform-tck/blob/10.0.0.Final/src/com/sun/ts/tests/servlet/api/jakarta_servlet/servletcontext/TestServlet.java#L195

Just that there are values. (empty set or null set are failures to the TCK)

**Comment by joakime** [comment]
And `getResourcePaths(String)` is not actually present in the SPEC document.

https://github.com/jakartaee/servlet/blob/6.0.0/spec/src/main/asciidoc/servlet-spec-body.adoc

**Comment by joakime** [comment]
The javadoc only shows an example with a trailing slash.

https://github.com/jakartaee/servlet/blob/6.0.0/api/src/main/java/jakarta/servlet/ServletContext.java#L172-L216

I think we should return the trailing slash, can't see why we would not.
Just amazed that this isn't in the TCK or SPEC.  (hey, we pass the Servlet 6.0 TCK on Jetty 12.0.0.beta3, we should be good, right?)

**Comment by joakime** [comment]
Opened PR #10085

**Comment by ppodgorsek** [comment]
Thank you for being so reactive, @joakime ! And thank you for the explanation, it's very good to know, I always thought the interface reflected the official spec. (that trailing `/` rule was already mentioned in the JavaDoc since Javax Servlet 2.3)

I'll definitely spend more time going through the TCK to learn more about it. :)

**Comment by joakime** [comment]
PR #10085 is now merged into `jetty-12.0.x` HEAD
